common/djangoapps/entitlements/management/commands/expire_old_entitlements.py

```diff
@@ -0,0 +1,76 @@
+"""
+Management command for expiring old entitlements.
+"""
+
+import logging
+
+from django.core.management import BaseCommand
+from django.core.paginator import Paginator
+
+from entitlements.models import CourseEntitlement
+from entitlements.tasks.v1.tasks import expire_old_entitlements
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+PAGE_SIZE = 10000  # arbitrary, should be adjusted if we see performance issues in the wild
+
+
+class Command(BaseCommand):
+    """
+    Management command for expiring old entitlements.
+
+    Most entitlements get expired as the user interacts with the platform,
+    because the LMS checks as it goes. But if the learner has not logged in
+    for a while, we still want to reap these old entitlements. So this command
+    should be run every now and then (probably daily) to expire old
+    entitlements.
+
+    The command's goal is to pass a narrow subset of entitlements to an
+    idempotent Celery task for further (parallelized) processing.
+    """
+    help = 'Expire old entitlements.'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-c', '--commit',
+            action='store_true',
+            dest='commit',
+            default=False,
+            help='Submit tasks for processing.'
+        )
+
+    def handle(self, *args, **options):
+        logger.info('Looking for entitlements which may be expirable.')
+
+        # This query could be optimized to return a more narrow set, but at a
+        # complexity cost. See bug LEARNER-3451 about improving it.
+        entitlements = CourseEntitlement.objects.filter(expired_at__isnull=True).order_by('id')
+
+        entitlements = Paginator(entitlements, PAGE_SIZE, allow_empty_first_page=False)
+
+        if options.get('commit'):
+            logger.info('Enqueuing entitlement expiration tasks for %d candidates.', entitlements.count)
+        else:
+            logger.info(
+                'Found %d candidates. To enqueue entitlement expiration tasks, pass the -c or --commit flags.',
+                entitlements.count
+            )
+            return
+
+        succeeded, failed = 0, 0
+        for page_num in entitlements.page_range:
+            try:
+                page = entitlements.page(page_num)
+                expire_old_entitlements.delay(page, logid=str(page_num))
+            except Exception:  # pylint: disable=broad-except
```

Comment:
I am generally not a fan of Broad exceptions.  Is there something that we expect to find here? Or is this just meant to be a catch all?

Reply:
The two broad exception catches are cargo-culted from another management command.  They are meant to be catch alls to keep the job running no matter what weirdness happens below them.

This specific one is probably less necessary than the other one (this one would likely only fail if we have celery problems? in which case trying another page is not much better than just waiting a day and trying again).  I'm happy to drop this catch line.

Reply:
Done.